tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/validation/TclValidatorTest.xtend

```diff
@@ -0,0 +1,121 @@
+package org.testeditor.tcl.dsl.validation
+
+import com.google.inject.Provider
+import javax.inject.Inject
+import org.eclipse.emf.common.util.URI
+import org.eclipse.xtext.junit4.util.ParseHelper
+import org.eclipse.xtext.junit4.validation.ValidationTestHelper
+import org.eclipse.xtext.resource.XtextResourceSet
+import org.junit.Before
+import org.junit.Test
+import org.testeditor.aml.AmlModel
+import org.testeditor.aml.dsl.AmlStandaloneSetup
+import org.testeditor.tcl.dsl.tests.parser.AbstractParserTest
+
+class TclValidatorTest extends AbstractParserTest {
+
+	@Inject
+	private Provider<XtextResourceSet> resourceSetProvider;
+
+	protected ParseHelper<AmlModel> amlParser
+
+	@Inject
+	ValidationTestHelper validator;
+
+	@Before
+	def void initializeAmlParseHelper() {
+		val amlInjector = new AmlStandaloneSetup().createInjectorAndDoEMFRegistration
+		amlParser = amlInjector.getInstance(ParseHelper)
+	}
+
+	@Test
+	def void validateStringArray() {
+		val aml = getAMLWithValueSpace('''#[ "New", "Open" ]''')
+		var tcl = getTCLWithValue("Test", "New")
+
+		val resourceSet = resourceSetProvider.get
+		amlParser.parse(aml, URI.createURI("swt.aml"), resourceSet)
+
+		var model = parser.parse(tcl, URI.createURI("Test.tcl"), resourceSet)
+		validator.assertNoIssues(model)
+
+		tcl = getTCLWithValue("Test2", "Save")
+		model = parser.parse(tcl, URI.createURI("Test2.tcl"), resourceSet)
+		assertFalse(validator.validate(model).isEmpty)
+	}
+
+	@Test
+	def void validateNumberRange() {
+		val aml = getAMLWithValueSpace("2 ... 5")
+		var tcl = getTCLWithValue("Test", "4")
+
+		val resourceSet = resourceSetProvider.get
+		amlParser.parse(aml, URI.createURI("swt.aml"), resourceSet)
+
+		var model = parser.parse(tcl, URI.createURI("Test.tcl"), resourceSet)
+		validator.assertNoIssues(model)
+
+		tcl = getTCLWithValue("Test2", "1")
+		model = parser.parse(tcl, URI.createURI("Test2.tcl"), resourceSet)
+		assertFalse(validator.validate(model).isEmpty)
```

Comment:
You may can show a example, how to create a reference error object to compare to?